app/controllers/api/v2/transfers_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    module Transfers
+      # Provides a JSON API controller for Transfers.
+      # See: http://jsonapi-resources.com/ for JSONAPI::Resource documentation.
+      class TransfersController < JSONAPI::ResourceController
+        # By default JSONAPI::ResourceController provides most of the standard behaviour.
+        # However in this case we want to redirect create and update operations to the correct polymorphic type.
+
+        def process_operations(operations)
+          # We need to determine the polymorphic type of the transfer to create based on any template provided.
+          operations.each do |operation|
+            context = operation.options[:context]
+            attributes = operation.options[:data][:attributes]
+
+            # Skip the operation if it does not contain a transfer template.
+            next unless attributes.key?(:transfer_template_uuid)
+
+            # Get the transfer template and use it to update the context and attributes.
+            template = TransferTemplate.with_uuid(attributes[:transfer_template_uuid]).first
+            context[:polymorphic_type] = template.transfer_class_name
+            attributes[:transfers] = template.transfers if template.transfers.present?
+
+            # Remove the UUID of the transfer template from the attributes.
+            attributes.delete(:transfer_template_uuid)
+          end
+
+          super(operations)
+        end
+      end
```

Comment:
Assuming that Processor calls create by passing the context in create_resource, this should work.

If resource_klass.new is called directly anywhere, it won't work though because operation's resource_klass is still TransferResource.